File: leetcode/src/python/pascals_triangle_2.py
from typing import List

class PascalsTriangle2:
    def getRow(self, rowIndex: int) -> List[int]:
        result = []
        if rowIndex == 0:
            result = [1]
        elif rowIndex == 1:
            result = [1, 1]
        else:
            last_list = [1, 1]
            for i in range(rowIndex - 1):
                current_list = [1]
                for j in range(len(last_list) - 1):
                    current_list.append(last_list[j] + last_list[j + 1])
                current_list.append(1)
                result = current_list
                last_list = current_list
        return result

if __name__ == '__main__':
    instance = PascalsTriangle2()
    rowIndex = 3
    print(instance.getRow(rowIndex))

File: leetcode/src/scala/PascalsTriangle2.scala
object PascalsTriangle2 {
    def generate(numRows: Int): List[List[Int]] = {
        var result = List(List(1))
        if (numRows >= 2) result = result :+ List(1, 1)
        if (numRows > 2) {
            var last = List(1, 1)
            Array.range(2, numRows).foreach(i => {
                val current = (Array(1) ++ Array.range(0, last.length - 1).collect({case j: Int => last.apply(j) + last.apply(j + 1)}) ++ Array(1)).toList
                result = result :+ current
                last = current
            })
        }
        result
    }
}